services/localfile/server/localfile.go

```diff
@@ -628,6 +630,39 @@ func (s *server) Rename(ctx context.Context, req *pb.RenameRequest) (*emptypb.Em
 	return &emptypb.Empty{}, nil
 }
 
+func (s *server) Readlink(ctx context.Context, req *pb.ReadlinkRequest) (*pb.ReadlinkReply, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	logger.Info("readlink", "filename", req.Filename)
+	if err := util.ValidPath(req.Filename); err != nil {
+		return nil, err
+	}
+	stat, err := os.Lstat(req.Filename)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "stat error: %v", err)
+	}
+	if stat.Mode()&os.ModeSymlink != os.ModeSymlink {
+		return nil, status.Errorf(codes.FailedPrecondition, "%v is not a symlink", req.Filename)
+	}
+	linkvalue, err := os.Readlink(req.Filename)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "readlink error: %v", err)
+	}
+	return &pb.ReadlinkReply{Linkvalue: linkvalue}, nil
+}
+
+func (s *server) Symlink(ctx context.Context, req *pb.SymlinkRequest) (*emptypb.Empty, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	logger.Info("symlink", "target", req.Target, "linkname", req.Linkname)
+	if err := util.ValidPath(req.Linkname); err != nil {
```

Comment:
do you also want to check req.Target?

Reply:
I skipped the check on purpose. I've added a comment explaining why.